test/inductor/test_torchinductor.py

```diff
@@ -386,9 +389,15 @@ def run(*ex, **kwargs):
             self.assertEqual(len(results_that_require_grad), 0)
         else:
             actual_grad = compute_grads(example_inputs, kwargs, actual, grads)
+
+            if reference_in_float:
+                expect_grad = reference_to_expect(actual_grad, correct_grad)
```

Comment:
why is it correct not to flatten/unflatten here? Don't the gradients have the same structure as the forward operation?

Reply:
`compute_grads` returns from `torch.autograd.grad(...)` which is always a flat tuple according to the [docstring](https://pytorch.org/docs/stable/generated/torch.autograd.grad.html).